File: java/src/util/List.java
package util;

public class List {
    public static ListNode create(int[] li) {
        ListNode pre = new ListNode(-1);
        ListNode p = pre;
        for (int i = 0; i < li.length; i++) {
            p.next = new ListNode(li[i]);
            p = p.next;
        }
        return pre.next;
    }

    public static void prettyPrint(ListNode head) {
        ListNode p = head;
        while (p != null) {
            System.out.print("->");
            System.out.print(p.val);
            p = p.next;
        }
        System.out.println();
    }

    public static void main(String[] args) {
        ListNode head = List.create(new int[]{1, 2, 3, 4, 5});
        List.prettyPrint(head);
    }
}

File: scala/src/util/List.scala
package util

object List {
  def create(li: Array[Int]): ListNode = {
    val pre = ListNode(-1)
    var p = pre
    li.map {
      item =>
        p.next = ListNode(item)
        p = p.next
    }
    pre.next
  }

  def prettyPrint(head: ListNode) {
    head match {
      case h: ListNode =>
        print("->" + head.value)
        prettyPrint(head.next)
      case _ => return
    }
  }

  def main(args: Array[String]) {
    val head = List.create(Array(1, 2, 3, 4, 5, 6, 7, 8, 9))
    List.prettyPrint(head)
  }
}
